### PR TITLE
Improvements to che-mount

### PR DESCRIPTION
**Pull request by l0rd**
- Included unison-fsmonitor to use unison `-repeat` option
  - More unison options: -copyonconflict -auto -prefer=newer -log=false
  - Added linux specific syntax to run che-mount in container

### CODE REVIEW

File: che-mount/sync.sh [hunk 1]
```diff
@@ -60,20 +74,35 @@ error_exit() {
...
+    exit 1
+fi
 info "INFO: ECLIPSE CHE: Successfully mounted user@$1:/projects"
- 
-while :
-do
-    unison /mntssh /mnthost -batch -fat -silent -auto -prefer=newer > /dev/null 2>&1
-    sleep 1
-done
+info "INFO: ECLIPSE CHE: First syncronisation is happening now and can take a long time. Please wait."
```
Comment (TylerJewell) [inline che-mount/sync.sh]: Lets simplify.  "INFO: ECLIPSE CHE: Initial sync... Please wait."

File: che-mount/sync.sh [hunk 2]
```diff
@@ -60,20 +74,35 @@ error_exit() {
...
+unison /mntssh /mnthost -batch -fat -silent -auto -prefer=newer -log=false > /dev/null 2>&1
+status=$?
+if [ $status -ne 0 ]; then
+    error "ERROR: Fatal error occurred ($status)"
+    exit 1
+fi
+info "INFO: ECLIPSE CHE: First syncronisation completed."
```
Comment (TylerJewell) [inline che-mount/sync.sh]: This line can be removed, as it will be clear from the next line.

File: che-mount/sync.sh [hunk 3]
```diff
@@ -60,20 +74,35 @@ error_exit() {
...
+info "INFO: ECLIPSE CHE: Now synchronization happens continuously every ${UNISON_REPEAT_DELAY_IN_SEC} seconds."
```
Comment (TylerJewell) [inline che-mount/sync.sh]: Change to "INFO: ECLIPSE CHE: Background sync continues every ${UNISON_REPEAT_DELAY_IN_SEC} seconds."

**Comment by TylerJewell** [comment]
We will need a matching improvement to the Che CLI with logic to handled invoking the container based upon the OS.  The following will need updates:
1: Che CLI (che.sh) in Che repo.
2: The che-mount docs explaining the nuances of the behavior.
3. The gist on the core Che web site, which outlines how to do Che mount initially.

File: che-mount/sync.sh (hunk 1 above)
Comment (l0rd) [inline che-mount/sync.sh]: Ok

File: che-mount/sync.sh (hunk 2 above)
Comment (l0rd) [inline che-mount/sync.sh]: Ok

File: che-mount/sync.sh (hunk 3 above)
Comment (l0rd) [inline che-mount/sync.sh]: Ok

**Comment by l0rd** [comment]
@TylerJewell I've updated this PR and created a new one for Che CLI: eclipse/che#2523

For the `che-mount` documentation do you prefer I just provide it here as a comment or "suggest edit" on https://eclipse-che.readme.io/docs/desktop-ide-mounting?

Where is the gist on the core Che web site?

**Comment by TylerJewell** [comment]
Ok - tested the changes - looks good to me.

**Comment by TylerJewell** [comment]
I tested the changes on windows with docker for windows and windows / boot2docker.

**Pull request merged**